src/services/contracts/deployments.ts

```diff
@@ -0,0 +1,87 @@
+import semverSatisfies from 'semver/functions/satisfies'
+import {
+  getSafeSingletonDeployment,
+  getSafeL2SingletonDeployment,
+  getMultiSendCallOnlyDeployment,
+  getFallbackHandlerDeployment,
+  getProxyFactoryDeployment,
+  getSignMessageLibDeployment,
+} from '@safe-global/safe-deployments'
+import type { SingletonDeployment, DeploymentFilter } from '@safe-global/safe-deployments'
+import type { ChainInfo, SafeInfo } from '@safe-global/safe-gateway-typescript-sdk'
+
+import { LATEST_SAFE_VERSION } from '@/config/constants'
+
+export const _tryDeploymentVersions = (
+  getDeployment: (filter?: DeploymentFilter) => SingletonDeployment | undefined,
+  network: string,
+  version: SafeInfo['version'],
+): SingletonDeployment | undefined => {
+  // Get deployments using LATEST_SAFE_VERSION for unsupported frontends/versions
+
+  return (
+    getDeployment({
+      version: version || LATEST_SAFE_VERSION,
+      network,
+    }) ||
+    getDeployment({
+      version: version || LATEST_SAFE_VERSION,
+    }) ||
+    getDeployment({
+      version: LATEST_SAFE_VERSION,
+      network,
+    }) ||
+    getDeployment({
+      version: LATEST_SAFE_VERSION,
+    })
+  )
+}
+
+export const _isLegacy = (safeVersion: SafeInfo['version']): boolean => {
+  const LEGACY_VERSIONS = '<=1.0.0'
+  return !!safeVersion && semverSatisfies(safeVersion, LEGACY_VERSIONS)
+}
+
+export const _isL2 = (chain: ChainInfo, safeVersion: SafeInfo['version']): boolean => {
+  const L2_VERSIONS = '>=1.3.0'
+
+  // We had L1 contracts on xDai, EWC and Volta so we also need to check version is after 1.3.0
+  return (
+    chain.l2 &&
+    semverSatisfies(
+      // Support unsupported versions
+      safeVersion || LATEST_SAFE_VERSION,
+      L2_VERSIONS,
+    )
```

Comment:
`|| LATEST_SAFE_VERSION` – again, some inline logic that seems innocuous but might shoot you in the foot.

If would instead do an early return like this:

```js
// Unsupported safe version
if (safeVersion === null) {
  return chain.l2
}
```